Automattic/happychat-client
# Add Unread Messages Button

This is the companion to 53355-gh-Automattic/wp-calypso

This PR adds a button that appears when the user is scrolled up, and new messages arrive. Clicking the button returns you to the latest message.

<img width="888" alt="Screen Shot 2021-06-03 at 11 39 31 PM" src="https://user-images.githubusercontent.com/518059/120746601-2ac3ab80-c4c5-11eb-987d-7f5c9974f2ff.png">

### Testing instructions

- Open a Happy Chat session
- Try lots of scenarios with scrolling to different positions while sending messages from both sides — the chat button should only appear when there are new messages you haven't seen yet from the operator.
- Check that the three Tracks events are firing correctly:
  - `happychatclient_unread_messages_button_show` when the button appears
  - `happychatclient_unread_messages_button_hide` when the button disappears (including when the button is clicked)
  - `happychatclient_unread_messages_button_click` when the button is clicked
